### ISSUE TITLE
Issue with Opener: Only Ring to Open is working not Open directly

### ISSUE THREAD

**Issue by riyas1980** [technyon/nuki_hub#310]
Hello,

First of all, thanks for your great work. I am trying nuki hub with my opener. I am using version 8.32

When added, my MQTT shows 2 entities (1 is enabled and 1 is disabled)

Intercom is enabled with Unlock as option
Unlatch is disabled

When I click on Unlock, it doesn't open my entrance door. The state change to Ring to Open. So If I ring my intercom, it's opening. 

I want to directly open the door without Ring to open. Is that possible ?

Thanks for your help. 

Kind Regards,

**Comment by riyas1980** [comment]
And I've enabled the unlatch button in Home Assistant. But when I press, it doesn't do anything

**Comment by iranl** [comment]
Unlock enables Ring to Open and lock disables Ring to Open..
Unlatch (equals Electronic Strike Actuation for the opener) opens a door.

The Unlatch button in Home Assistant only works when unlatch is enabled in the Nuki Hub configuration (Nuki Configuration -> Advanced Nuki Configuration). Set access level to "Full"

![image](https://github.com/technyon/nuki_hub/assets/25727444/b6a9da13-ab23-4060-bf4d-db98b7680e48)

**Comment by riyas1980** [comment]
> Unlock enables Ring to Open and lock disables Ring to Open. Unlatch (equals Electronic Strike Actuation for the opener) opens a door.
> 
> The Unlatch button in Home Assistant only works when unlatch is enabled in the Nuki Hub configuration (Nuki Configuration -> Advanced Nuki Configuration). Set access level to "Full"
> 
Hi,

Thanks for your reply. Actually it is already in Full Access level. But When I press the Unlatch button, Opener doesn't do anything. But in logbook, we can see that : "Unlatch changed to February 28, 2024 at 12:28 PM triggered by service Button: Press 12:28:42 PM - 21 hours ago".

For your information, when MQTT discovered Nuki Hub, only "Unlock" button was visible. Unlatch was disabled by default. I've enabled it but unfortunately it's not working. 

![Nuki_settings](https://github.com/technyon/nuki_hub/assets/35848952/9e8d5e62-9fe6-433b-adb1-6bf582e6b01a)

**Comment by iranl** [comment]
Unlatch button is disabled in HA by default. This is by design as a security measure.

The Home Assistant log will probably not be very helpfull here as it is only able to tell if it published something to the MQTT topic, what Nuki Hub does or doesn't do based on this can not really be discerned from within Home Assistant.

Just to make sure: You are able to unlatch/open your door without ringing using the Nuki smartphone app?

**Comment by riyas1980** [comment]
> Unlatch button is disabled in HA by default. This is by design as a security measure.
> 
> The Home Assistant log will probably not be very helpfull here as it is only able to tell if it published something to the MQTT topic, what Nuki Hub does or doesn't do based on this can not really be discerned from within Home Assistant.
> 
> Just to make sure: You are able to unlatch/open your door without ringing using the Nuki smartphone app?

Yes I am able to unlatch/open my door without ringing from my Nuki app. I really don't know what's wrong with my HA and Nuki Hub

**Comment by iranl** [comment]
Have you tried setting `nukiopener/lock/action` to `electricStrikeActuation` manually (using an app like MQTT Explorer)? 
To rule out an issue with Home Assistant. 
Please also check `nukiopener/lock/action` changes to `electricStrikeActuation` when pushing the Home Assistant "Unlatch" button.

What does `nukiopener/lock/completionStatus` and `nukiopener/lock/commandResult` show after setting action to `electricStrikeActuation`?
And does the door open when setting action to `electricStrikeActuation` manually?

**Comment by riyas1980** [comment]
> Have you tried setting `nukiopener/lock/action` to `electricStrikeActuation` manually (using an app like MQTT Explorer)? To rule out an issue with Home Assistant. Please also check `nukiopener/lock/action` changes to `electricStrikeActuation` when pushing the Home Assistant "Unlatch" button.
> 
> What does `nukiopener/lock/completionStatus` and `nukiopener/lock/commandResult` show after setting action to `electricStrikeActuation`? And does the door open when setting action to `electricStrikeActuation` manually?

Thanks for your help. I am out of town. Tomorrow I'll try and inform you. Thanks again

**Comment by riyas1980** [comment]
I am trying to change the setting to electricStrikeActuation with MQTT Explorer but I don't know where exactly I need to change. I am sorry. I am new to MQTT.

![Nuki_Edit](https://github.com/technyon/nuki_hub/assets/35848952/2cb81895-1acb-4c67-b7ed-bc621562dfa2)

I've even updated the Nuki Hub to 8.33 and I can see the continuous mode also. 

With MQTT explorer, this is what I get, when I click on unlock (action : ack)

![Nuki_Unlock](https://github.com/technyon/nuki_hub/assets/35848952/97dda0a2-3a7c-4ce3-ab46-e325648a0264)

But when I click on Unlatch, I have this message (action = Unknown_action): 

![Nuki_Unlatch1](https://github.com/technyon/nuki_hub/assets/35848952/c5a3a26c-ef8f-44d5-b53b-27306faed633)

In the history, I have 2 lines : First one is Unlatch and second one is Unknown_action

![Nuki_Unlatch2](https://github.com/technyon/nuki_hub/assets/35848952/1c0d0fb2-5498-4739-abca-04b5aa15b09d)

I don't know if my message is clear to you. I don't know how to edit the setting to "electricStrikeActuation"

Thanks again for your help

**Comment by iranl** [comment]
There is a bug in the code that exposes the "Unlatch" button in Home Assistant for the opener.

When using a Opener Home Assistant should not sent a "unlatch" command but the "electricStrikeActuation" command.

PR #316 should fix this bug.

You can download the binary that is created by GitHub with these changes from https://github.com/technyon/nuki_hub/actions/runs/8147478237/artifacts/1296509546

Please try the binary and report back if your issue is solved. You will know Home Assistant has picked up the changes if the name of the "Unlatch" button has changed to "Open".

**Comment by technyon** [comment]
In MQTT explorer, you should set lock/action to unlatch. When correct, Nuki Hub will respond with "ack" to the same topic.

**Comment by iranl** [comment]
Good to hear it works as expected now.

Nuki Hub is known to have varying delay of a couple of seconds between issueing a command and it's execution.
This is due to the nature of the bluetooth communication between the ESP and the lock/opener.
